wrapper.js

```diff
@@ -68,6 +68,28 @@ function setupMethods (soljson) {
   }
 
   var compile = function (input, optimise, readCallback) {
+    // NOTE: horribly inefficient
+    if (typeof input === 'string') {
+      var isStandardJSON = false;
+      try {
+        input = JSON.parse(input);
+        if (typeof input['language'] === 'string') {
+          isStandardJSON = true;
+        }
+      } catch (e) {
+      }
+
+      if (isStandardJSON) {
+        // NOTE: takes second argument as "readCallback"
+        return JSON.parse(compileStandardWrapper(input, optimise));
```

Comment:
`compileStandardWrapper` is only assigned in line 83 / 105

Reply:
Javascript scoping... 😉

Reply:
Ah, it's a callback!